libs/perun/components/src/lib/members-list/members-list.component.ts

```diff
@@ -270,4 +278,27 @@ export class MembersListComponent implements OnChanges, AfterViewInit {
 
     this.dialog.open(MemberTreeViewDialogComponent, config);
   }
+
+  getConsentsForMembers(members: RichMember[]): MemberWithConsentStatus[] | RichMember[] {
+    const result: MemberWithConsentStatus[] = [];
+    if (this.displayedColumns.includes('consentStatus')) {
+      members.forEach((member) => {
+        const mwc: MemberWithConsentStatus = member;
+        mwc.consent = this.translate.instant(
+          'CONSENTS.STATUS_' + this.transformConsentForMember(member, this.consents)
+        ) as string;
+        result.push(mwc);
+      });
+      return result;
+    }
+    return members;
+  }
+
+  transformConsentForMember(member: RichMember, consents: Consent[]): string {
+    if (!consents || consents.length === 0) return;
+    const membersConsents = consents.filter((consent) => consent.userId === member.userId);
+    // return latest consent (one with the highest id)
+    if (membersConsents.length === 0) return;
+    return membersConsents.reduce((a, b) => (a.id > b.id ? a : b)).status;
+  }
```

Comment:
The solution is nice but I have couple of suggestions. It would be nice to allow filtering/sorting by consents. However that would require to make the transformation every time the filter or sorting changes (also apply the `mat-sort-header` to the column) which is redundant. I would also still prefer if this was not handled by the table itself.

I suggest to pass the `MemberWithConsentStatus` from the parent since it has all the information.

It seems the `transformConsentForMember` is just a duplication of `ConsentStatusIconPipe`. You can use the pipe inside of `.ts` files as well. For this provide the pipe inside of  `@Component` decorator. In this case `providers: [ConsentStatusIconPipe]`. Than you can create an instance of the pipe like this `constructor(private somePipe: SomePipe)` and use it's transform method where needed `this.somePipe.transform(...)`.

Reply:
I also agree with the need for filtering. The most optimal solution would IMO be to use the dynamic members list and do the filtering backend side, however, that would require changes to the paginated members get method (which I'm not quite sure is necessary, depending on how large the list of assigned members can get, up to debate maybe?).
For now, I would be satisfied with the filtering taking place in resource-assigned-members component, which would also allow us to pass MemberWithConsentStatus to the list component (as Adam suggested).